src/SprykerEco/Zed/Heidelpay/Business/Encrypter/AesEncrypter.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+/**
+ * MIT License
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerEco\Zed\Heidelpay\Business\Encrypter;
+
+use SprykerEco\Zed\Heidelpay\HeidelpayConfig;
+
+class AesEncrypter implements EncrypterInterface
+{
+
+    const CYPHER_METHOD = 'aes-256-cbc';
+    const INIT_VECTOR_SEPARATOR = ':::';
+
+    /**
+     * @var \SprykerEco\Zed\Heidelpay\HeidelpayConfig
+     */
+    protected $config;
+
+    /**
+     * @param \SprykerEco\Zed\Heidelpay\HeidelpayConfig $config
+     */
+    public function __construct(HeidelpayConfig $config)
+    {
+        $this->config = $config;
+    }
+
+    /**
+     * @param string $data
+     *
+     * @return string
+     */
+    public function encryptData($data)
+    {
+        $encryptionKey = $this->config
+            ->getEncryptionKey();
+        $initVector = $this->getRandomPseudoBytes();
+
+        $encryptedData = openssl_encrypt(
+            $data,
+            static::CYPHER_METHOD,
+            $encryptionKey,
+            OPENSSL_RAW_DATA,
+            $initVector
+        );
+
+        return implode([$encryptedData, base64_encode($initVector)], static::INIT_VECTOR_SEPARATOR);
```

Comment:
Are you sure about the order of arguments? ;) IMO this cant work as separator is the first.

Reply:
You are right, fixed.